core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -376,66 +384,171 @@ public List<Acl> getPrefixedAclsSOT(String env, int tenantId) {
+   *     view)
+   * @param requestor The UserName of the person who is requesting the list of topics
+   * @param status Filter the response by the status of the requests e.g.
+   *     all,created,declined,deleted,approved
+   * @param showRequestsOfAllTeams boolean true if a superadmin with the permission to view
+   *     requests. otherwise false
+   * @param tenantId The identifier of the tenancy with which the requestor is from.
+   * @param teamId filter the results by the name of the team who created the request
+   * @param env filter the results by the environment identifier e.g. 1,2,3,4. Applied only when
+   *     allReqs is true
+   * @param wildcardSearch filter the results by a wildcard search of the topicnames. Applied only
+   *     when allReqs is true
+   * @return A list of TopicRequests that meet the filtered criteria. Applied only when allReqs is
+   *     true
+   */
+  public List<TopicRequest> getFilteredTopicRequests(
       boolean allReqs,
       String requestor,
       String status,
       boolean showRequestsOfAllTeams,
-      int tenantId) {
-    log.debug("selectTopicRequests {} {}", requestor, status);
-
-    Integer teamSelected = selectUserInfo(requestor).getTeamId();
-    List<TopicRequest> topicRequests;
-
-    if (showRequestsOfAllTeams) {
-      topicRequests = topicRequestsRepo.findAllByTenantId(tenantId);
-    } else {
-      topicRequests = topicRequestsRepo.findAllByTeamIdAndTenantId(teamSelected, tenantId);
+      int tenantId,
+      Integer teamId,
+      String env,
+      String wildcardSearch) {
+    if (log.isDebugEnabled()) {
+      log.debug(
+          "selectTopicRequests {} {} {} {} {} {}",
+          showRequestsOfAllTeams,
+          requestor,
+          status,
+          teamId,
+          env,
+          wildcardSearch);
     }
+    Integer teamSelected = selectUserInfo(requestor).getTeamId();
+    List<TopicRequest> topicRequests = new ArrayList<>();
+    List<TopicRequest> topicRequestListSub;
 
     if (allReqs) { // approvers
-
+      // Team Name is null here as it is the team who created the requested
+      // Description includes the teamId as that is how Claim topics know who is the owning team.
       List<TopicRequest> claimTopicReqs =
-          topicRequestsRepo.findAllByTopictypeAndTenantId(
-              RequestOperationType.CLAIM.value, tenantId);
-      topicRequests =
-          topicRequests.stream()
-              .filter(
-                  topicRequest ->
-                      !RequestOperationType.CLAIM.value.equals(topicRequest.getTopictype()))
-              .collect(Collectors.toList());
+          Lists.newArrayList(
+              findTopicRequestsByExample(
+                  RequestOperationType.CLAIM.value,
+                  null,
+                  env,
+                  status,
+                  tenantId,
+                  String.valueOf(teamSelected)));
+
+      // remove any claims created by your team.
       claimTopicReqs =
```

Comment:
these requests would not have the requests created by your team, as they are created by other team.
if 'description' field has team id, then they are claim requests to be approved

Reply:
Ah I see what you mean so this can be removed completely.